packages/tools/devtools/devtools-core/src/data-visualization/DefaultVisualizers.ts

```diff
@@ -103,6 +108,34 @@ export const visualizeSharedCell: VisualizeSharedObject = async (
 	}
 };
 
+/**
+ * Wrapper class for {@link DataObject} to provide a {@link ISharedDirectory} root.
+ * @remarks Intended for devtools internal use only.
+ */
+export class VisualDataObject extends DataObject {
+	public override get root(): ISharedDirectory {
+		return super.root;
+	}
+}
+
+/**
+ * Default {@link VisualizeSharedObject} for {@link DataObject}.
+ */
+export const visualizeDataObject: VisualizeDataObject = async (
+	dataObject: DataObject,
+	visualizeChildData: VisualizeChildData,
+): Promise<FluidObjectTreeNode> => {
+	const dataObjectRoot = (dataObject as VisualDataObject).root;
```

Comment:
I was going to say maybe we want runtime checks here to make sure what we got is actually a `DataObject`, but it feels a bit redundant if we check it enough when registering the visualizer, so maybe just add a comment here that we don't do further checks because we already validated when we registered the visualizer function?